python/ray/autoscaler/_private/autoscaler.py

```diff
@@ -1478,6 +1479,36 @@ def summary(self) -> Optional[AutoscalerSummary]:
             pending_resources=pending_resources,
         )
 
+    def decorate_load_metrics_summary(self, lm_summary: LoadMetricsSummary) -> None:
+        """In place update fields like `node_type_mapping` with optional
+        information from the node provider.
+
+        Args:
+          lm_summary: The object to update in place.
+
+        """
+
+        # For type checking, assert that this object has been instantitiated.
+        assert self.provider
+
+        if not self.non_terminated_nodes:
+            return None
+
+        # The field is only None for compatibility purposes. It should be
+        # guaranteed to be filled out here.
+        assert lm_summary.usage_by_node is not None
+
+        lm_summary.node_type_mapping = {}
+
+        for node_id in self.non_terminated_nodes.all_node_ids:
+            ip = self.provider.internal_ip(node_id)
```

Comment:
Probably there should be some try/catches around these node provider calls -- even the existing summary logic could use that -- failures in node provider calls used for observability info shouldn't interrupt the autoscaler.

Reply:
hmmm i think practically this call hits a cached entry in every(?) node provider implementation, but on a more philosophical note: if we treat observability as a first class requirement, isn't it reasonable for a failure to mean the autoscaler is in a degraded state if we can't explain its actions?

Reply:
> practically this call hits a cached entry in every(?) node provider implementation

I've observed otherwise based on user complaints.
The "cacheing" should be inherent to the interface, but it isn't.

> isn't it reasonable for a failure to mean the autoscaler is in a degraded state if we can't explain its actions?

Empirically, it's better to always assume the autoscaler is in a degraded state and to do your best to defend against that.

Reply:
I see you rewrote the logic to cache the IPs in the autoscaler code itself -- good idea, I approve.
Also, the logic is simpler now, nice.